pkg/operator/ceph/object/controller.go

```diff
@@ -203,12 +213,24 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 	// DELETE: the CR was deleted
 	if !cephObjectStore.GetDeletionTimestamp().IsZero() {
 		logger.Debugf("deleting store %q", cephObjectStore.Name)
+		cephObjectStore.Spec.PreservePoolsOnDelete = false
 		cfg := clusterConfig{context: r.context, store: cephObjectStore}
 		err := cfg.deleteStore()
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "failed to delete store %q. ", cephObjectStore.Name)
 		}
 
+		// TODO: Add cleanupPolicy for confirming deletion of object buckets.
+		// TODO: Always set PreservePoolsOnDelete to false if object buckets cleanupPolicy is present.
+		if !cephObjectStore.Spec.PreservePoolsOnDelete {
+			// Wait for all the object buckets in an object store to be deleted
+			err = r.waitForObjectBucketCleanup(cephObjectStore)
```

Comment:
Although if the cleanupPolicy is set, we really shouldn't wait for object bucket cleanup since the intent is to destroy everything.